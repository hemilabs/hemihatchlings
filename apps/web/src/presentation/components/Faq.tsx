import plus from '../../assets/plus.svg'
import minus from '../../assets/minus.svg'
import HowItWorks from './HowItWorks'
import { useState } from 'react'

function Faq(): JSX.Element {
  const [showHistory, setShowHistory] = useState(false)

  const toogleHistory = () => {
    setShowHistory(!showHistory)
  }

  return (
    <div className="flex flex-col items-center gap-8">
      <div className="w-full max-w-screen-xl items-center p-8 border border-white">
        <div onClick={toogleHistory}
          className="flex flex-row justify-between">
          <h2 className="font-space font-bold text-white text-lg
            sm:text-xl md:text-2xl lg:text-3xl xl:text-4xl">
            HOW IT WORKS
          </h2>

          <img className={`${showHistory ? 'hidden' : ''} w-4 md:w-6 lg:w-8`}
            src={plus}
            alt='Plus Icon' />

          <img className={`${showHistory ? '' : 'hidden'} w-4 md:w-6 lg:w-8`}
            src={minus}
            alt='Minus Icon' />
        </div>
        <div className={`${showHistory ? '' : 'hidden'}`}>
          <HowItWorks />
        </div>
      </div>
      <div className="w-full max-w-screen-xl items-center p-8 border border-white">
        <a className='flex flex-row justify-between' href="https://github.com/hemilabs" target='_blank'>
          <h2 className="font-space font-normal text-white text-lg
            sm:text-xl md:text-2xl lg:text-3xl xl:text-4xl">
            ADD A FEATURE
          </h2>
          <img src={plus} className="w-4 md:w-6 lg:w-8" alt="Plus Icon" />
        </a>
      </div>
      <div className=" w-full max-w-screen-xl items-center p-8
        border border-white">
        <a className='flex flex-row justify-between' href="https://discord.gg/RyhaPp7NvQ" target='_blank'>
          <h2 className="font-space font-normal text-white text-lg
            sm:text-xl md:text-2xl lg:text-3xl xl:text-4xl">
            COMMUNITY
          </h2>
          <img src={plus} className="w-4 md:w-6 lg:w-8 " alt="Plus Icon" />
        </a>
      </div>
    </div>
  )
}

export default Faq
