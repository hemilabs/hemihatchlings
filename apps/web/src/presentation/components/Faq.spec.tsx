import { describe, it, expect} from 'vitest'
import React from 'react'
import { render } from '@testing-library/react'
import Faq from './Faq'

describe('src/presentation/Faq', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Faq />)

    expect(asFragment()).toMatchSnapshot()
  })
})
