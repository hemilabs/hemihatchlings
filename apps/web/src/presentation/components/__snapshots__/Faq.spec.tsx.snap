// Vitest Snapshot v1

exports[`src/presentation/Faq > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center gap-8"
  >
    <div
      class="w-full max-w-screen-xl items-center p-8 border border-white"
    >
      <div
        class="flex flex-row justify-between"
      >
        <h2
          class="font-space font-bold text-white text-lg sm:text-xl md:text-2xl lg:text-3xl xl:text-4xl"
        >
          HOW IT WORKS
        </h2>
        <img
          alt="Plus Icon"
          class=" w-4 md:w-6 lg:w-8"
          src="/src/assets/plus.svg"
        />
        <img
          alt="Minus Icon"
          class="hidden w-4 md:w-6 lg:w-8"
          src="/src/assets/minus.svg"
        />
      </div>
      <div
        class="hidden"
      >
        <div
          class="font-space font-light text-white text-sm mt-4 sm:text-base md:text-lg lg:text-xl"
        >
          <p
            class="py-4"
          >
            Hemi Hatchlings offers an immersive introduction to the Hemi Network's approach to scaling the Ethereum and Bitcoin networks, merging elemental lore with blockchain finality principles. Here's how the story unfolds:
          </p>
          <h3
            class="font-space font-bold text-white text-md py-4 sm:text-xl md:text-lg lg:text-xl xl:text-2xl"
          >
            ü•ö Choose Your Elemental Egg
          </h3>
          <p
            class="py-4"
          >
            Kick off your journey by selecting an egg (transaction) endowed with elemental power:
          </p>
          <ul
            class="ml-8 py-4"
          >
            <li>
              üî• Fire
            </li>
            <li>
              üíß Water
            </li>
            <li>
              üåç Earth 
            </li>
          </ul>
          <p
            class="py-4"
          >
            This choice sets the stage for your journey‚Ä¶
          </p>
          <div
            class="border border-b my-4"
          />
          <h3
            class="font-space font-bold text-white text-md py-4 sm:text-xl md:text-lg lg:text-xl xl:text-2xl"
          >
            Ethereum's Role: The Journey of Growing from Hatchling Into an Adolescent
          </h3>
          <ul
            class="ml-2 md:ml-8"
          >
            <li
              class="py-4"
            >
              <p
                class="font-bold py-4"
              >
                1. üê£ The Magic Moment of Hatching (Transaction):
              </p>
              <p>
                As soon as you select your egg, the magic is set in motion (a transaction is created), and your elemental egg instantly transforms into a baby hatchling. It must prepare to travel into the mystical forest (the Hemi network).
              </p>
            </li>
            <li
              class="py-4"
            >
              <p
                class="font-bold py-4"
              >
                2. üßô‚Äç‚ôÇÔ∏è The Guidance of the Wise Sage (Hemi Sequencer):
              </p>
              <p>
                The wise sage of the forest oversees the baby hatchlings, ensuring they are ready for their journey. The sage groups the hatchlings together, organizing their paths (rollup) toward the next stage of their journey.
              </p>
            </li>
            <li
              class="py-4"
            >
              <p
                class="font-bold py-4"
              >
                3. üõ°Ô∏è The Forest Guardians (Validators):
              </p>
              <p>
                As your hatchling continues, it is guided towards the celestial sanctuary (Ethereum). Here, the legendary Forest Guardians (Ethereum validators), who have proven their worth and commitment by offering a part of their magic (staking their ETH), watch over the hatchling. Their role is to keep it safe and ensure everything runs smoothly. They rely on the integrity of the wise sage's preparations, assuming that the hatchlings meet all necessary criteria unless proven otherwise during a challenge period (a story for another time).
              </p>
            </li>
            <li
              class="py-4"
            >
              <p
                class="font-bold py-4"
              >
                4. ‚ú® The Blessing of the Guardians (Gathering Consensus):
              </p>
              <p>
                Before the hatchling can grow further, it requires the consensus of other guardians. They gather around the hatchling, each adding their approval (attestations). This is like a magical chant that ensures everything is in order.
              </p>
            </li>
            <li
              class="py-4"
            >
              <p
                class="font-bold py-4"
              >
                5. üê• Growing to Adolescence (Finality):
              </p>
              <p>
                Once consensus is reached and the guardians have given their approval, the hatchling transforms into the next phase. This moment‚Äîwhen your hatchling becomes an adolescent‚Äîis magical and irreversible, forever a part of the forest's history.
              </p>
            </li>
          </ul>
          <p
            class="py-4"
          >
            In the enchanting forest, Ethereum's finality ensures that the transformation of your hatchling into an adolescent is significant. Just as guardians protect the forest, Ethereum's validators secure the network, guaranteeing every hatchling's growth is an important moment recorded for all time.
          </p>
          <div
            class="border border-b my-4"
          />
          <h3
            class="font-space font-bold text-white text-md py-4 sm:text-xl md:text-lg lg:text-xl xl:text-2xl"
          >
            Bitcoin's Role: The Hatchling's Transition to Adulthood
          </h3>
          <ul
            class="ml-2 md:ml-8"
          >
            <li
              class="py-4"
            >
              <p
                class="font-bold py-4"
              >
                1. üåå Adventuring Beyond the Forest:
              </p>
              <p>
                As our adolescent grows, it embarks on a quest beyond the magical forest (the Ethereum network) and into the vast mountains (the Bitcoin network). Here, the terrain is governed by ancient miners (miners) who delve deep into the mountain's heart, extracting magical crystals (solving cryptographic puzzles) to forge the path ahead, one step (block) at a time.
              </p>
            </li>
            <li
              class="py-4"
            >
              <p
                class="font-bold py-4"
              >
                2. ‚õèÔ∏è The Ancient Miners (Miners):
              </p>
              <p>
                These tireless workers have dedicated their lives to the mountains, using their skills and tools to unearth rare and powerful crystals. Their work is crucial, as each crystal they find and validate contributes to constructing a mighty fortress (blockchain) where all the land's most precious treasures (transactions) are kept safe from dark forces.
              </p>
            </li>
            <li
              class="py-4"
            >
              <p
                class="font-bold py-4"
              >
                3. üåü Growing Up:
              </p>
              <p>
                The adolescent ventures through these mountains, seeking the blessings of the ancient miners. Each blessing (block confirmation) is like a rite of passage, strengthening the adolescent and preparing it for the final transformation. The journey is arduous, and many blessings are needed for it to mature into an adult fully.
              </p>
            </li>
            <li
              class="py-4"
            >
              <p
                class="font-bold py-4"
              >
                4. üîÆ Gathering the Crystals:
              </p>
              <p>
                To support their transformation into adulthood, adolescents must gather magical crystals (confirmations). Each crystal imbues them with strength and wisdom. This process symbolizes the network's rigorous verification method, ensuring that each step in its journey is secure and irreversible.
              </p>
            </li>
            <li
              class="py-4"
            >
              <p
                class="font-bold py-4"
              >
                5. üå± Adulthood (Bitcoin Finality):
              </p>
              <p>
                After receiving enough blessings from the ancient miners, a spectacular event unfolds: The adolescent ascends, emerging in full adulthood. This ascension marks a significant milestone celebrated throughout the lands. The adolescent emergence is a testament to the journey's success, permanently recorded in the caves of the mountain fortress!
              </p>
            </li>
          </ul>
          <p
            class="py-4"
          >
            In weaving together Ethereum and Bitcoin finality, our mystical journey underscores the critical role these two blockchain realms play in guiding and securing the lifecycle of each hatchling.
          </p>
          <p
            class="py-4"
          >
            With their vigilant oversight, Ethereum's guardians ensure that every hatchling becoming an adolescent is recorded with permanence. Then, Bitcoin's ancient miners extend and secure this protection, forging a path of growth and transformation celebrated across both lands (networks).
          </p>
          <p
            class="py-4"
          >
            Together, Ethereum's proof-of-stake mechanism and Bitcoin's proof-of-work consensus act as twin pillars of strength, ensuring that from the moment of adolescence to the ascent into adulthood, every milestone is engraved on-chain.
          </p>
        </div>
      </div>
    </div>
    <div
      class="w-full max-w-screen-xl items-center p-8 border border-white"
    >
      <a
        class="flex flex-row justify-between"
        href="https://github.com/hemilabs"
        target="_blank"
      >
        <h2
          class="font-space font-normal text-white text-lg sm:text-xl md:text-2xl lg:text-3xl xl:text-4xl"
        >
          ADD A FEATURE
        </h2>
        <img
          alt="Plus Icon"
          class="w-4 md:w-6 lg:w-8"
          src="/src/assets/plus.svg"
        />
      </a>
    </div>
    <div
      class=" w-full max-w-screen-xl items-center p-8 border border-white"
    >
      <a
        class="flex flex-row justify-between"
        href="https://discord.gg/hemixyz"
        target="_blank"
      >
        <h2
          class="font-space font-normal text-white text-lg sm:text-xl md:text-2xl lg:text-3xl xl:text-4xl"
        >
          COMMUNITY
        </h2>
        <img
          alt="Plus Icon"
          class="w-4 md:w-6 lg:w-8 "
          src="/src/assets/plus.svg"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
